lib/net/parser.js

```diff
@@ -136,13 +136,20 @@ class Parser extends EventEmitter {
       return null;
     }
 
-    // Count length of the cmd.
+    // check for null padding and find length of the command
     let i = 0;
-    for (; data[i + 4] !== 0 && i < 12; i++);
-
-    if (i === 12) {
-      this.error('Non NULL-terminated command.');
-      return null;
+    for (; i < 12; i++) {
+      if (data[i + 4] === 0) {
+        let j = i;
+        while (j < 12) {
+          if (data[j + 4] !== 0) {
+            this.error('Invalid null padding.');
+            return null;
+          }
+          j++;
+        }
+        break;
+      }
```

Comment:
No, I don't think this will work as this leaves the null padding at the end of the buffer which results in the cmd string having `\x00` at the end of the string. For example `hello` with null padding will be parsed as `hello\x00\x00\x00\x00\x00\x00`

Reply:
```js
'use strict';

function parse(data) {
  let end = 16;
  for (let i = 4; i < 16; i++) {
    if (data[i] === 0) {
      end = i;
    } else {
      if (end < 16)
        throw new Error('Invalid null padding.');
    }
  }

  const cmd = data.toString('ascii', 4, end);
  console.log(cmd);
}

function write(str) {
  const buf = Buffer.alloc(16, 0x00);
  const cmd = Buffer.from(str, 'ascii');
  cmd.copy(buf, 4);
  return buf;
}

function writeInvalid(str) {
  const buf = Buffer.alloc(16, 0x00);
  const cmd = Buffer.from(str, 'ascii');
  cmd.copy(buf, 4);
  buf[14] = 0x00;
  buf[15] = 0xff;
  return buf;
}

// hello
parse(write('hello'));

// 123456789012
parse(write('123456789012'));

// Invalid null padding
parse(writeInvalid('hello'));

```